src/routes/users.ts

```diff
@@ -24,20 +30,32 @@ router.get("/:id", isAuth, isCurrentUser, getUsersById);
 
 router.get("/:id/subjects", isAuth, isCurrentUser, getSubjectsOfUser);
 
-// router.get("/:id/grades", getAllGrades);
+router.get("/:id/grades", isAuth, isCurrentUser, getAllGrades);
 
-router.delete("/:id", isAuth, isAdmin, deleteUserById);
+router.get("/:id/grades/:idGrade", isAuth, isCurrentUser, getGradeById);
 
-// router.delete("/:id/subjects/:idSubject", deleteSubjectsOfUser);
+router.delete("/:id", isAuth, isAdmin, deleteUserById);
 
-// router.delete("/:id/grades/:idGrade", deleteGradeById);
+router.delete(
+  "/:id/subjects/:idSubject",
+  isAuth,
+  isAdmin,
+  deleteSubjectsOfUser
+);
+
+router.delete(
+  "/:id/grades/:idGrade",
+  isAuth,
+  isAtLeastTeacher,
+  deleteGradeById
+);
 
-// router.post("/:id/subjects", addSubjectsOfUser);
+router.post("/:id/subjects", addSubjectsOfUser);
 
-// router.post("/:id/grades", addGradeOfUser);
+router.post("/:id/grades", isAuth, isAdmin, addGradeOfUser);
```

Comment:
isAdmin o isAtLeastTeacher?

Reply:
Poichè stiamo gestendo lato client solamente l'Admin, per adesso ho lasciato così, in realtà dovrebbe essere solamente del teacher di quello studente oppure l'admin, questo middleware non ancora esiste, pertanto riserviamoci d'implementarlo in futuro.